src/components/ManageGallery/OrganizeCollection/Editor/CollectionEditor.tsx

```diff
@@ -118,6 +119,26 @@ function CollectionEditor({ queryRef, onValidChange }: Props) {
     [stagedCollectionState, onValidChange]
   );
 
+  // Check if the collection has unsaved changes
+  const lastStagedCollection = useRef<StagedCollection | Record<string, unknown>>({});
```

Comment:
Open to any idea on what's the best way to check unsaved changes

Reply:
If I understand this logic correctly, the following issues may arise

1. Add something to a collection
2. Remove thing from collection
3. Still get prompted as if I made changes.  

I like the `cloneDeep` stuff but I think we just want to store the initial collection value in a ref and compare against that for the rest of their editing

Reply:
Thanks terence!


https://user-images.githubusercontent.com/4480258/200010036-eacb081c-55d0-47fc-a6ac-101609c23308.mp4

Now the issue has been resolved. The initial collection value will be in `ref` and there is a `useEffect` that monitor it with `stagedCollections`

Reply:
🔥🔥🔥